File: algorithms/0448.find_all_numbers_disappeared_in_an_array/find_all_numbers_disappeared_in_an_array.rs
impl Solution {
    pub fn find_disappeared_numbers(nums: Vec<i32>) -> Vec<i32> {
        let mut numbers = nums.clone();
        for &n in nums.iter() {
            numbers[n as usize - 1] = 0;
        }
        let mut result = vec![];
        for i in 0..nums.len() {
            if numbers[i] != 0 {
                result.push(i as i32 + 1);
            }
        }
        result
        // numbers.iter().map(|&x| x).filter(|x| *x != 0).collect()
    }
}

File: algorithms/0448.find_all_numbers_disappeared_in_an_array/find_all_numbers_disappeared_in_an_array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> result;
        for (const int &e: nums) {
            const int n = e > 0 ? e : -e;
            const int num = nums[n - 1] > 0 ? nums[n-1] : -nums[n-1];
            nums[n - 1] = -num;
        }
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] > 0) {
                result.push_back(i + 1);
            }
        }
        return result;
    }
};